docs/selectors.rst

```diff
@@ -0,0 +1,104 @@
+.. _selectors:
+
+Selectors
+=========
+
+The selector is an internal structure used in the client. Given an array of
+connections, it chooses a single connection. Some selectors don't even worry
+much about the internals of the connection. There are some in-built selectors
+that you can use or you can even write and use your own custom selector.
+
+.. note:: A selector is called every time a request to the Elasticsearch
+          server is to be made. The list of all available connections are
+          passed to the selector.
+
+In-Built Selectors
+------------------
+
+These selectors are defined inside `elasticsearch.selectors` module. There are
+three of them:
+
+* **RoundRobinSelector (Default)**: The connections are selected in a round robin
+  fashion. i.e. #1 connection will be chosen on the first request, #2
+  connection will be chosen on the second request and so on. This ensures a
+  nearly even load across each node in the cluster.
+
+* **StickyRoundRobinSelector**: This selector will always return('stick') the same
```

Comment:
I believe there is a difference between the connection data structure we create in es-lua as an abstraction to what a connection is and a real connection to the server. In regards to our abstraction, yes we can use the same connection. However, the real connection to the server is always a new one when we execute `http.request(request)`. So the idea is good, but the implementation with luasocket does not allow `his case is ideal for persistent connections where a considerable time is spent in opening and closing connections.` to be true.